src/DataLayouts/cuda.jl

```diff
@@ -92,3 +92,14 @@ function Base.copyto!(
     end
     return dest
 end
+
+function Base.copyto!(
+    dest::VF{S},
+    bc::Union{VF{S, A}, Base.Broadcast.Broadcasted{VFStyle{A}}},
+) where {S, A <: CUDA.CuArray}
+    _, _, _, Nv, Nh = size(bc)
```

Comment:
How are we querying `Nf` here? Should `Nh` be one?

Reply:
In general, `1D` threadblocks, with appropriate `slab` calls when direct indexing is not perimitted, may be more efficient!
```
_max_threads_cuda() = 256

function _configure_threadblock(nitems)
    nthreads = min(_max_threads_cuda(), nitems)
    nblocks = cld(nitems, nthreads)
    return (nthreads, nblocks)
end

function Base.copyto!(
    dest::VF{S},
    bc::Union{VF{S, A}, Base.Broadcast.Broadcasted{VFStyle{A}}},
) where {S, A <: CUDA.CuArray}
    nitems = length(parent(dest))
    nthreads, nblocks = _configure_thread_blocks(nitems)
    if nitems > 0
        CUDA.@cuda threads = (nthreads,) blocks = (nblocks) knl_copyto_vf!(dest, bc)
    end
    return dest
end

function knl_copyto_vf!(dest, src)
    gidx = CUDA.threadIdx().x + (CUDA.blockIdx().x - 1) * CUDA.blockDim().x
    _, _, _, Nv, _ = size(dest)
    f = cld(gidx, Nv)
    v = gidx - Nv * (f - 1)
    dest_slab = slab(dest, f)
    src_slab = slab(src, f)
    @inbounds dest_slab[v] = src_slab[v]
    return nothing
end
```

Reply:
If operators are applies to fields (`f`s) recursively, then we may only need to distribute threads based on `Nv`!

Reply:
> How are we querying `Nf` here? Should `Nh` be one?

We don't need `Nf` (each thread can access all fields), and yes, `Nh == 1`

I agree this is inefficient (we're only doing a fraction of the work we should be doing)

Reply:
Actually, i have an idea on how to fix this.

Reply:
See #1216

Reply:
I realise what we have is far from optimal, but we can clean this up once we have other pieces in.

Reply:
Sure. We can revisit this once we have other pieces in.